pkg/controllermanager/controller/controllerregistration/seed/add.go

```diff
@@ -360,3 +360,17 @@ func (r *Reconciler) MapControllerDeploymentToAllSeeds(ctx context.Context, log
 
 	return nil
 }
+
+func shootNetworkingTypeHasChanged(old, new *gardencorev1beta1.Networking) bool {
+	if old == nil && new == nil {
+		return false
+	}
+	if old == nil && new != nil {
+		// if new is non-nil then return true if new has a type set
+		return new.Type != nil
+	}
+	if old != nil && new == nil {
```

Comment:
Shouldn't it be like this to be precise?

```suggestion
	if old != nil && old.Type != nil && new == nil {
```

Reply:
You are right, Thanks!

Reply:
It should rather be,
```
	if old != nil && new == nil {
		// if old was non-nil and had a type set, return true
		return old.Type != nil
	}
```
Else if `old!=nil` but `old.Type!=nil` and `new==nil`, there could be a nil pointer error.